pkg/util/client/client.go

```diff
@@ -31,12 +31,47 @@ func CreatePatch(before, after client.Object) (client.Patch, error) {
 	return client.RawPatch(patchBase.Type(), patchBytes), nil
 }
 
-func Patch(ctx context.Context, c client.Client, before, after client.Object) error {
-	patch, err := CreatePatch(before, after)
+// Patch applies the merge patch of client.Object with SSA.
+// If strict is true, the resourceVersion will be part of the patch, make this call fail if
+// client.Object was changed.
+func Patch(ctx context.Context, c client.Client, obj client.Object, strict bool, update func() (bool, error)) error {
```

Comment:
```suggestion
func Patch(ctx context.Context, c client.Client, obj client.Object, update func() (bool, error)) error {
```

Is there any reasons that we still need to provide the `strict` option?

Reply:
Yes, due to https://github.com/kubernetes-sigs/kueue/pull/2553#discussion_r1670493486.

Reply:
I see. So, we can remove the `strict` option from the `PatchStatus`, right?

Reply:
Removed. Thanks.